e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js

```diff
@@ -256,9 +257,31 @@ describe(
       });
 
       getDashboardCard().within(() => {
-        cy.findByText("0").should("be.visible");
+        cy.findByText("Sleep card").should("be.visible");
       });
     });
+
+    it("should preserve filter value when navigating between the dashboard and the question", () => {
+      createDashboardWithSlowCard();
+      cy.get("@dashboardId").then(visitDashboard);
+
+      filterWidget().findByPlaceholderText("sleep").type("5{enter}");
+
+      getDashboardCard().within(() => {
+        cy.findByText("Sleep card").click();
+        cy.wait("@card");
+      });
+
+      filterWidget().findByPlaceholderText("sleep").should("have.value", "5");
+
+      queryBuilderHeader().findByLabelText("Back to Sleep dashboard").click();
+
+      getDashboardCard().within(() => {
```

Comment:
Could you also check that data for the card has been re-requested?

Reply:
added check on number of requests, didn't find a beautiful way to verify the actual request params